### PR TITLE
DOC: Update outdated caveats for Anaconda and HTML parsing (#9032)

### LINKED ISSUE #9032: DOC: cleanup outdated caveats for HTML parsing (Anaconda LXML)
Anaconda now ships with lxml 3.4.0:
- http://docs.continuum.io/anaconda/pkg-docs.html

http://pandas.pydata.org/pandas-docs/stable/gotchas.html#html-table-parsing

> Issues with using Anaconda
> 
> Anaconda ships with lxml version 3.2.0; the following workaround for Anaconda was successfully used to deal with the versioning issues surrounding lxml and BeautifulSoup4.

Is this still a issue, now that Anaconda ships with lxml 3.4.0? (e.g. is there a test for this)

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #9032

### CODE REVIEW

**Comment by mroeschke** [comment]
True. I personally haven't had installation issues with those packages with anaconda. Should be okay to delete this section.

File: doc/source/gotchas.rst [hunk 1]
```diff
@@ -514,57 +514,6 @@ parse HTML tables in the top-level pandas io function ``read_html``.
...
-         pip install 'git+git://github.com/lxml/lxml.git'
-
-         # install the latest version of beautifulsoup4
-         pip install 'bzr+lp:beautifulsoup'
-
-      Note that you need `bzr <http://bazaar.canonical.com/en>`__ and `git
-      <http://git-scm.com>`__ installed to perform the last two operations.
-
-.. |svm| replace:: **strictly valid markup**
-.. _svm: http://validator.w3.org/docs/help.html#validation_basics
```
Comment (jorisvandenbossche) [inline doc/source/gotchas.rst]: All those references are used above as well, so can't be removed (except for anaconda I think, you should check)

**Review by jorisvandenbossche — COMMENTED** [review]
I agree this can probably removed altogether. Just a comment on the references

**Comment by mroeschke** [comment]
Thanks for the catch @jorisvandenbossche. Accidentally deleted those. Looks like Anaconda isn't referenced anywhere else on that page.

**Comment by jorisvandenbossche** [comment]
@mroeschke Thanks!

**Pull request merged**